Index out of range error when censoring cyrillic language

### Describe the bug

When i try to censor a word in cyrillic it throw a panic error

### What do you see?

```go
panic: runtime error: index out of range [8] with length 8

goroutine 62 [running]:

github.com/TwiN/go-away.(*ProfanityDetector).Censor(0xc00039bd28, {0xc0001a0c10, 0x10})
        /home/user/.gvm/pkgsets/go1.18rc1/global/pkg/mod/github.com/!twi!n/go-away@v1.6.0/goaway.go:179 +0x6d1
```

### What do you expect to see?

A correct censorship of the word

### List the steps that must be taken to reproduce this issue

```go
filter := goaway.NewProfanityDetector().WithCustomDictionary(profanities, fp, fn)

cleanedWord := f.Censor("крывавы")
```

### Version

1.6.0

### Additional information

_No response_

--- a/goaway.go
+++ b/goaway.go
@@ -149,26 +149,6 @@ func (g *ProfanityDetector) indexToRune(s string, index int) int {
 	return -1
 }
 
-func (g *ProfanityDetector) takeRunesFromIndex(s string, index int) string {
-	ret := make([]rune, 0)
-	for i, i2 := range []rune(s) {
-		if i >= index {
-			ret = append(ret, i2)
-		}
-	}
-	return string(ret)
-}
-
-func (g *ProfanityDetector) takeRunesToIndex(s string, index int) string {
-	ret := make([]rune, 0)
-	for i, i2 := range []rune(s) {
-		if i < index {
-			ret = append(ret, i2)
-		}
-	}
-	return string(ret)
-}
-
 // Censor takes in a string (word or sentence) and tries to censor all profanities found.
 func (g *ProfanityDetector) Censor(s string) string {
 	censored := []rune(s)

--- a/goaway.go
+++ b/goaway.go
@@ -171,7 +171,7 @@ func (g *ProfanityDetector) takeRunesToIndex(s string, index int) string {
 
 // Censor takes in a string (word or sentence) and tries to censor all profanities found.
 func (g *ProfanityDetector) Censor(s string) string {
-	censored := s
+	censored := []rune(s)
 	var originalIndexes []int
 	s, originalIndexes = g.sanitize(s, true)
 	// Check for false negatives
@@ -180,10 +180,8 @@ func (g *ProfanityDetector) Censor(s string) string {
 		for currentIndex != -1 {
 			if foundIndex := strings.Index(s[currentIndex:], word); foundIndex != -1 {
 				for i := 0; i < len([]rune(word)); i++ {
-					runeIndex := g.indexToRune(censored, currentIndex+foundIndex+i)
-					prefix := g.takeRunesToIndex(censored, originalIndexes[runeIndex])
-					postfix := g.takeRunesFromIndex(censored, originalIndexes[runeIndex]+1)
-					censored = prefix + "*" + postfix
+					runeIndex := g.indexToRune(string(censored), currentIndex+foundIndex+i)
+					censored[originalIndexes[runeIndex]] = '*'
 				}
 				currentIndex += foundIndex + len([]rune(word))
 			} else {
@@ -211,10 +209,8 @@ func (g *ProfanityDetector) Censor(s string) string {
 		for currentIndex != -1 {
 			if foundIndex := strings.Index(s[currentIndex:], word); foundIndex != -1 {
 				for i := 0; i < len([]rune(word)); i++ {
-					runeIndex := g.indexToRune(censored, currentIndex+foundIndex+i)
-					prefix := g.takeRunesToIndex(censored, originalIndexes[runeIndex])
-					postfix := g.takeRunesFromIndex(censored, originalIndexes[runeIndex]+1)
-					censored = prefix + "*" + postfix
+					runeIndex := g.indexToRune(string(censored), currentIndex+foundIndex+i)
+					censored[originalIndexes[runeIndex]] = '*'
 				}
 				currentIndex += foundIndex + len([]rune(word))
 			} else {
@@ -222,7 +218,7 @@ func (g *ProfanityDetector) Censor(s string) string {
 			}
 		}
 	}
-	return censored
+	return string(censored)
 }
 
 func (g ProfanityDetector) sanitize(s string, rememberOriginalIndexes bool) (string, []int) {

--- a/goaway.go
+++ b/goaway.go
@@ -138,6 +138,37 @@ func (g *ProfanityDetector) ExtractProfanity(s string) string {
 	return ""
 }
 
+func (g *ProfanityDetector) indexToRune(s string, index int) int {
+	count := 0
+	for i := range s {
+		if i == index {
+			return count
+		}
+		count++
+	}
+	return -1
+}
+
+func (g *ProfanityDetector) takeRunesFromIndex(s string, index int) string {
+	ret := make([]rune, 0)
+	for i, i2 := range []rune(s) {
+		if i >= index {
+			ret = append(ret, i2)
+		}
+	}
+	return string(ret)
+}
+
+func (g *ProfanityDetector) takeRunesToIndex(s string, index int) string {
+	ret := make([]rune, 0)
+	for i, i2 := range []rune(s) {
+		if i < index {
+			ret = append(ret, i2)
+		}
+	}
+	return string(ret)
+}
+
 // Censor takes in a string (word or sentence) and tries to censor all profanities found.
 func (g *ProfanityDetector) Censor(s string) string {
 	censored := s
@@ -148,10 +179,13 @@ func (g *ProfanityDetector) Censor(s string) string {
 		currentIndex := 0
 		for currentIndex != -1 {
 			if foundIndex := strings.Index(s[currentIndex:], word); foundIndex != -1 {
-				for i := 0; i < len(word); i++ {
-					censored = censored[:originalIndexes[foundIndex+currentIndex+i]] + "*" + censored[originalIndexes[foundIndex+currentIndex+i]+1:]
+				for i := 0; i < len([]rune(word)); i++ {
+					runeIndex := g.indexToRune(censored, currentIndex+foundIndex+i)
+					prefix := g.takeRunesToIndex(censored, originalIndexes[runeIndex])
+					postfix := g.takeRunesFromIndex(censored, originalIndexes[runeIndex]+1)
+					censored = prefix + "*" + postfix
 				}
-				currentIndex += foundIndex + len(word)
+				currentIndex += foundIndex + len([]rune(word))
 			} else {
 				break
 			}
@@ -162,8 +196,9 @@ func (g *ProfanityDetector) Censor(s string) string {
 		currentIndex := 0
 		for currentIndex != -1 {
 			if foundIndex := strings.Index(s[currentIndex:], word); foundIndex != -1 {
-				originalIndexes = append(originalIndexes[:foundIndex+currentIndex], originalIndexes[foundIndex+len(word):]...)
-				currentIndex += foundIndex + len(word)
+				runeIndex := g.indexToRune(s, currentIndex+foundIndex)
+				originalIndexes = append(originalIndexes[:runeIndex], originalIndexes[runeIndex+len(word):]...)
+				currentIndex += foundIndex + len([]rune(word))
 			} else {
 				break
 			}
@@ -175,10 +210,13 @@ func (g *ProfanityDetector) Censor(s string) string {
 		currentIndex := 0
 		for currentIndex != -1 {
 			if foundIndex := strings.Index(s[currentIndex:], word); foundIndex != -1 {
-				for i := 0; i < len(word); i++ {
-					censored = censored[:originalIndexes[foundIndex+currentIndex+i]] + "*" + censored[originalIndexes[foundIndex+currentIndex+i]+1:]
+				for i := 0; i < len([]rune(word)); i++ {
+					runeIndex := g.indexToRune(censored, currentIndex+foundIndex+i)
+					prefix := g.takeRunesToIndex(censored, originalIndexes[runeIndex])
+					postfix := g.takeRunesFromIndex(censored, originalIndexes[runeIndex]+1)
+					censored = prefix + "*" + postfix
 				}
-				currentIndex += foundIndex + len(word)
+				currentIndex += foundIndex + len([]rune(word))
 			} else {
 				break
 			}
@@ -213,7 +251,7 @@ func (g ProfanityDetector) sanitize(s string, rememberOriginalIndexes bool) (str
 	}
 	var originalIndexes []int
 	if rememberOriginalIndexes {
-		for i, c := range s {
+		for i, c := range []rune(s) {
 			if c != ' ' {
 				originalIndexes = append(originalIndexes, i)
 			}

--- a/goaway_test.go
+++ b/goaway_test.go
@@ -100,6 +100,10 @@ func TestProfanityDetector_Censor(t *testing.T) {
 			expectedCensoredOutput: "But the table is on ****ing fire",
 		},
 		{
+			input:                  "““““““““““““But the table is on fucking fire“",
+			expectedCensoredOutput: "““““““““““““But the table is on ****ing fire“",
+		},
+		{
 			input:                  "f.u_ck this s.h-i~t",
 			expectedCensoredOutput: "*.*_** this *.*-*~*",
 		},

